### PR TITLE
Resolves #286. 

### LINKED ISSUE #286: RFC: Crud bake theme
I just had the idea that a Crud bake theme could be created. It would be very basic and essentially just have empty controller templates.

Then you could generate your controllers using bake.

Would this be quicker than using 'File > New class'? I guess that depends on your IDE.

### PR DESCRIPTION
**Pull request by davidyell**
Added a controller bake template.

### CODE REVIEW

File: src/Template/Bake/Controller/controller.ctp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * Allows templating of Controllers generated from bake.
+ *
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
```
Comment (ADmad) [inline src/Template/Bake/Controller/controller.ctp]: Nope, Crud plugin isn't copyrighted by CSF :)

Comment (davidyell) [inline src/Template/Bake/Controller/controller.ctp]: Whoops!

File: src/Template/Bake/Controller/controller.ctp [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+foreach ($components as $component):
+    $classInfo = $this->Bake->classInfo($component, 'Controller/Component', 'Component');
+%>
+ * @property <%= $classInfo['fqn'] %> $<%= $classInfo['name'] %>
+<% endforeach; %>
+ */
+class <%= $name %>Controller extends AppController
+{
+<%
+echo $this->Bake->arrayProperty('helpers', $helpers, ['indent' => false]);
```
Comment (ADmad) [inline src/Template/Bake/Controller/controller.ctp]: I don't want to promote using these properties. Helpers should be loaded in View::initialize() and components in Controller::initialize()

Comment (davidyell) [inline src/Template/Bake/Controller/controller.ctp]: What are the thoughts of others on this? Should the templates deviate from the default bake templates?

Comment (lorenzo) [inline src/Template/Bake/Controller/controller.ctp]: I also prefer if people use AppView for loading helpers

Comment (ADmad) [inline src/Template/Bake/Controller/controller.ctp]: > Should the templates deviate from the default bake templates?

Maintaining consistency with default bake templates is not necessary. FOC is more opinionated which it was it also has it's own app skeleton.

**Comment by davidyell** [comment]
Please can someone review for me, and let me know if there are any other changes required.

**Comment by jadb** [comment]
LGTM @davidyell, can you squash please?

**Comment by davidyell** [comment]
Sure, I'll do that tomorrow morning, er, later this morning. Eish 1am!

**Comment by davidyell** [comment]
Job done.

**Comment by GuidoHendriks** [comment]
Great job! :+1:

**Pull request merged**